std/functional.d

```diff
@@ -992,10 +992,11 @@ template memoize(alias fun)
         alias Args = Parameters!fun;
         import std.typecons : Tuple;
 
-        static ReturnType!fun[Tuple!Args] memo;
-        auto t = Tuple!Args(args);
+        static ReturnType!fun[ulong] memo;
+        auto t = Tuple!Args(args).hashOf;
```

Comment:
This is the wrong fix because you won't be able to resolve collisions, e.g. two calls with different `args` to memoize can end up overwriting the same record in the hash table, just because `tuple(args1).hashOf == tuple(args2).hashOf`, while `args1 != args2`.